src/rviz/default_plugin/markers/text_view_facing_marker.cpp

```diff
@@ -57,6 +57,8 @@ void TextViewFacingMarker::onNewMessage(const MarkerConstPtr& old_message, const
 {
   ROS_ASSERT(new_message->type == visualization_msgs::Marker::TEXT_VIEW_FACING);
 
+  if(new_message->text.empty()|| new_message->text.find_first_not_of(" \t\n\v\f\r") == std::string::npos)
```

Comment:
I cannot "hide" the text anymore by simply displaying an empty string?

Reply:
> I cannot "hide" the text anymore by simply displaying an empty string?

You never could. With an empty text field (at least after something has been published there), the plugin segfaults.
With whitespace-only strings, OGRE triggers an exception:

> 1550484180.013238148: ros.rviz: OGRE EXCEPTION(7:InternalErrorException): Vertex Buffer: Out of memory in GLHardwareVertexBuffer::lock at /usr/src/ogre-v1-9-0/RenderSystems/GL/src/OgreGLHardwareVertexBuffer.cpp (line 124)
terminate called after throwing an instance of 'Ogre::InternalErrorException'

That being said, it remains underspecified semantics whether empty-string should be allowed to disable a marker or not.
We went ahead and forbade it because there are already a number of other pitfalls that can make you publish markers that never show in Rviz.
If you prefer the alternative semantics to allow empty-string to serve as an alternative to `color.a=0` or `action=DELETE`, this request can be changed accordingly.

Reply:
OK. Could you please address the other comments?